### PR TITLE
[NOD-1145] Normalize panics in flows

### PR DESCRIPTION
**Pull request by stasatdaglabs**

### CODE REVIEW

File: protocol/flows/blockrelay/handle_relay_invs.go [hunk 1]
```diff
@@ -169,25 +169,23 @@ func (flow *handleRelayInvsFlow) readMsgBlock() (
...
 
 func (flow *handleRelayInvsFlow) processAndRelayBlock(requestQueue *hashesQueueSet, block *util.Block) error {
-
 	blockHash := block.Hash()
 	isOrphan, isDelayed, err := flow.DAG().ProcessBlock(block, blockdag.BFNone)
 	if err != nil {
 		// When the error is a rule error, it means the block was simply
 		// rejected as opposed to something actually going wrong, so log
-		// it as such. Otherwise, something really did go wrong, so panic.
+		// it as such. Otherwise, something really did go wrong, so return
```
Comment (someone235) [inline protocol/flows/blockrelay/handle_relay_invs.go:183]: Maybe remove this comment?

File: protocol/flows/handshake/handshake.go [hunk 2]
```diff
@@ -93,10 +99,7 @@ func HandleHandshake(context HandleHandshakeContext, router *routerpkg.Router,
 
 	err = context.AddToPeers(peer)
 	if err != nil {
-		if errors.Is(err, common.ErrPeerWithSameIDExists) {
-			return nil, false, err
-		}
-		panic(err)
+		return nil, false, err
```
Comment (someone235) [inline protocol/flows/handshake/handshake.go:102]: ErrPeerWithSameIDExists should be a protocol error

File: protocol/flows/relaytransactions/relaytransactions.go [hunk 3]
```diff
@@ -188,10 +188,10 @@ func (flow *handleRelayedTransactionsFlow) receiveTransactions(requestedTransact
 			// When the error is a rule error, it means the transaction was
 			// simply rejected as opposed to something actually going wrong,
 			// so log it as such. Otherwise, something really did go wrong,
-			// so panic.
+			// so return the error as is.
```
Comment (someone235) [inline protocol/flows/relaytransactions/relaytransactions.go:191]: Change comment

File: protocol/flows/handshake/handshake.go [hunk 4]
```diff
@@ -109,7 +112,7 @@ func HandleHandshake(context HandleHandshakeContext, router *routerpkg.Router,
 
 	err = router.RemoveRoute([]wire.MessageCommand{wire.CmdVersion, wire.CmdVerAck})
 	if err != nil {
-		panic(err)
+		return nil, false, err
```
Comment (someone235) [inline protocol/flows/handshake/handshake.go:115]: it should panic here

Comment (stasatdaglabs) [reply protocol/flows/handshake/handshake.go:116]: We spoke about this and decided to leave it as is.

File: protocol/manager.go [hunk 5]
```diff
@@ -57,10 +59,24 @@ func (m *Manager) IBDPeer() *peerpkg.Peer {
...
 
 // AddBlock adds the given block to the DAG and propagates it.
 func (m *Manager) AddBlock(block *util.Block, flags blockdag.BehaviorFlags) error {
-	return m.context.AddBlock(block, flags)
+	err := m.context.AddBlock(block, flags)
+	if err != nil {
+		if protocolErr := &(protocolerrors.ProtocolError{}); errors.As(err, &protocolErr) {
+			return err
+		}
+		panic(err)
```
Comment (svarogg) [inline protocol/manager.go:79]: Please add "TODO(libp2p): Remove panic once RPC is integrated into protocol architecture"

**Pull request merged**